qualtran/pyzx_interop/bloq_to_pyzx_circuit.py

```diff
@@ -0,0 +1,179 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Iterable
+
+import numpy as np
+import pyzx as zx
+from attrs import define
+from numpy.typing import NDArray
+
+from qualtran import Bloq, CompositeBloq, LeftDangle, Register, RightDangle, Signature, Soquet
+
+ZXQubitMap = dict[str, NDArray[np.integer]]
+"""A mapping from register names to an NDArray of ZX qubits"""
+
+
+@define
```

Comment:
can this be frozen? if yes: make it so; if no, use `attrs.mutable` and explain why in docstring